client/state/plugins/installed/reducer.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import uniqBy from 'lodash/uniqBy';
+import omit from 'lodash/omit';
+import findIndex from 'lodash/findIndex';
+/**
+ * Internal dependencies
+ */
+import status from './status/reducer';
+import {
+	PLUGINS_RECEIVE,
+	PLUGINS_REQUEST,
+	PLUGINS_REQUEST_SUCCESS,
+	PLUGINS_REQUEST_FAILURE,
+	PLUGIN_ACTIVATE_REQUEST_SUCCESS,
+	PLUGIN_DEACTIVATE_REQUEST_SUCCESS,
+	PLUGIN_UPDATE_REQUEST_SUCCESS,
+	PLUGIN_AUTOUPDATE_ENABLE_REQUEST_SUCCESS,
+	PLUGIN_AUTOUPDATE_DISABLE_REQUEST_SUCCESS,
+	PLUGIN_INSTALL_REQUEST_SUCCESS,
+	PLUGIN_REMOVE_REQUEST_SUCCESS,
+	SERIALIZE,
+	DESERIALIZE
+} from 'state/action-types';
+
+/*
+ * Tracks the requesting state for installed plugins on a per-site index.
+ */
+export function isRequesting( state = {}, action ) {
+	switch ( action.type ) {
+		case PLUGINS_REQUEST:
+			return Object.assign( {}, state, { [ action.siteId ]: true } );
+		case PLUGINS_RECEIVE:
+			return Object.assign( {}, state, { [ action.siteId ]: false } );
+		case SERIALIZE:
+		case DESERIALIZE:
+			return {};
+		default:
+			return state;
+	}
+}
+
+/*
+ * Tracks the requesting state for installed plugins on a per-site index.
+ */
+export function hasRequested( state = {}, action ) {
```

Comment:
What will this be used for? To avoid re-requesting plugins for a site if they've already been received? Typically we'd be liberal in allowing a query component to re-request in its lifecycle methods regardless of whether it'd been requested in the past.